adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -244,18 +256,20 @@ func (swagger *MgwSwagger) SetXWso2ProductionEndpointMgwSwagger(productionURL st
 	return errors.New("invalid production endpoint")
 }
 
-func (swagger *MgwSwagger) setXWso2PrdoductionEndpoint() error {
-	xWso2APIEndpoints := getXWso2Endpoints(swagger.vendorExtensions, productionEndpoints)
+func (swagger *MgwSwagger) setXWso2ProductionEndpoint() error {
+	xWso2APIEndpoints, err := getXWso2Endpoints(swagger.vendorExtensions, productionEndpoints)
 	if xWso2APIEndpoints != nil && len(xWso2APIEndpoints) > 0 {
 		swagger.productionUrls = xWso2APIEndpoints
-	} else {
+	} else if err != nil {
 		return errors.New("error encountered when extracting endpoints")
```

Comment:
Better of we can provide more details about the endpoint url and etc

Reply:
If we propagate the error generated from `getHostNamePortBasepath` method it contains those information right? (rather than generating a new error)

Reply:
The `getHostPortandBasePath` method logs the endpoint that causes the issue. This is only an intermediate step that is used to propagate the error. Which API in particular, is logged in the parent call (i.e. api name, version, id, etc.). Parent call as in `ApplyAPIProjectFromAPIM` or `ApplyAPIProjectInStandaloneMode`. This particular error is not logged anywhere. Do we still need to include all those details?

Please refer to the following adapter log when a malformed endpoint is used.

![APIM_AdapterCrashFix](https://user-images.githubusercontent.com/30475839/124806822-bb851f80-df7a-11eb-97b1-b1620d1916df.png).